p2p/receiver/handler.go

```diff
@@ -190,6 +193,47 @@ func (srv *Handler) SendAnchoredDocument(ctx context.Context, docReq *p2ppb.Anch
 	return &p2ppb.AnchorDocumentResponse{Accepted: true}, nil
 }
 
+// HandleGetDocument handles HandleGetDocument message
+
+func (srv *Handler) HandleGetDocument(ctx context.Context, peer peer.ID, protoc protocol.ID, msg *p2ppb.Envelope) (*pb.P2PEnvelope, error) {
+	m := new(p2ppb.GetDocumentRequest)
+	err := proto.Unmarshal(msg.Body, m)
+	if err != nil {
+		return convertToErrorEnvelop(err)
+	}
+
+	res, err := srv.GetDocument(ctx, m)
+	if err != nil {
+		return convertToErrorEnvelop(err)
+	}
+
+	nc, err := srv.config.GetConfig()
+	if err != nil {
+		return convertToErrorEnvelop(err)
+	}
+
+	p2pEnv, err := p2pcommon.PrepareP2PEnvelope(ctx, nc.GetNetworkID(), p2pcommon.MessageTypeGetDocRep, res)
+	if err != nil {
+		return convertToErrorEnvelop(err)
+	}
+
+	return p2pEnv, nil
+}
+
+// GetDocument receives document identifier and retrieves the corresponding CoreDocument from the repository
+func (srv *Handler) GetDocument(ctx context.Context, docReq *p2ppb.GetDocumentRequest) (*p2ppb.GetDocumentResponse, error) {
```

Comment:
Let's move this to the above function itself and rename `HandleGetDocument` -> `GetDocuemnt` ?

Reply:
That logic would differ from how the other methods on the handlers work (SendAnchoredDoc, ReqDocumentSignature). I agree there is redundancy here, I have noted this in a separate issue #651. Do you think it makes sense to address that in a separate PR?